src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -502,31 +487,46 @@ public boolean isSwiftStorage() {
     }
     
     //This function applies to a single datafile
-    public boolean isSwiftStorage(FileMetadata metadata){
+    private boolean isSwiftStorage(FileMetadata metadata){
         if (metadata.getDataFile().getStorageIdentifier().startsWith("swift://")) {
             return true;
         }
         return false;
     }
     
+    
+    private Boolean showComputeButtonForDataset = null;
     //This function applies to an entire dataset
     public boolean showComputeButton() {
+        if (showComputeButtonForDataset != null) {
+            return showComputeButtonForDataset;
+        }
+        
         if (isSwiftStorage() && (settingsService.getValueForKey(SettingsServiceBean.Key.ComputeBaseUrl) != null)) {
-            return true;
+            showComputeButtonForDataset = true;
+        } else {
+            showComputeButtonForDataset = false;
         }
-        return false;
+        return showComputeButtonForDataset;
```

Comment:
Why did you change this? At the very least, you should, change it to `showComputeButtonForDataset =(isSwiftStorage() && (settingsService.getValueForKey(SettingsServiceBean.Key.ComputeBaseUrl) != null))
return showComputeButtonForDataset`

Reply:
But that's exactly what it's doing now, no? 
The reason I changed it in the first place was to only run the test once, then cache the result and return it the next time the page asks. (The problem with this method was that it was implemented in a super expensive way; AND the page calls it 24 times (!) as it's loading).

Reply:
the only difference is a useless if else statement. Rather than saying `if x{y=true}else{y=false}`, just having 'y=x` is both simpler and faster.

Reply:
I use verbose, multi-line if / else because it's easier to read. I know it's subjective.